client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts

```diff
@@ -1,124 +1,66 @@
-const mapApiError = ( error: any ) => {
-	return {
-		body: {
-			code: error.code,
-			message: error.message,
-			data: error.data,
-		},
-		status: error.status,
-	};
-};
-
 export const useCredentialsForm = ( onSubmit: () => void ) => {
-	const translate = useTranslate();
 	const importSiteQueryParam = useQuery().get( 'from' ) || '';
+	const {
+		isPending,
+		mutate: requestAutomatedMigration,
+		error,
+		isSuccess,
+		variables,
+	} = useSiteMigrationCredentialsMutation();
 
-	const fieldMapping = {
-		from_url: {
-			fieldName: 'siteAddress',
-			errorMessage: translate( 'Enter a valid URL.' ),
-		},
-		username: {
-			fieldName: 'username',
-			errorMessage: translate( 'Enter a valid username.' ),
-		},
-		password: {
-			fieldName: 'password',
-			errorMessage: translate( 'Enter a valid password.' ),
-		},
-		migration_type: {
-			fieldName: 'howToAccessSite',
-			errorMessage: null,
-		},
-		notes: {
-			fieldName: 'notes',
-			errorMessage: null,
-		},
-	};
-
-	const setGlobalError = ( message?: string | null | undefined ) => {
-		// eslint-disable-next-line @typescript-eslint/no-use-before-define
-		setError( 'root', {
-			type: 'manual',
-			message: message ?? translate( 'An error occurred while saving credentials.' ),
-		} );
-	};
-
-	const handleMigrationError = ( err: MigrationError ) => {
-		let hasUnmappedFieldError = false;
-
-		if ( err.body?.code === 'rest_invalid_param' && err.body?.data?.params ) {
-			Object.entries( err.body.data.params ).forEach( ( [ key ] ) => {
-				const field = fieldMapping[ key as keyof typeof fieldMapping ];
-				const keyName =
-					// eslint-disable-next-line @typescript-eslint/no-use-before-define
-					'backup' === accessMethod && field?.fieldName === 'siteAddress'
-						? 'backupFileLocation'
-						: field?.fieldName;
-
-				if ( keyName ) {
-					const message = field?.errorMessage ?? translate( 'Invalid input, please check again' );
-					// eslint-disable-next-line @typescript-eslint/no-use-before-define
-					setError( keyName as keyof CredentialsFormData, { type: 'manual', message } );
-				} else if ( ! hasUnmappedFieldError ) {
-					hasUnmappedFieldError = true;
-					setGlobalError();
-				}
-			} );
-		} else {
-			setGlobalError( err.body?.message );
-		}
-	};
-
-	const { isPending, requestAutomatedMigration } = useSiteMigrationCredentialsMutation( {
-		onSuccess: () => {
-			recordTracksEvent( 'calypso_site_migration_automated_request_success' );
-			onSubmit();
-		},
-		onError: ( error: any ) => {
-			handleMigrationError( mapApiError( error ) );
-			recordTracksEvent( 'calypso_site_migration_automated_request_error' );
-		},
-	} );
+	const serverSideError = useFormErrorMapping( error, variables );
 
 	const {
 		formState: { errors },
 		control,
 		handleSubmit,
 		watch,
-		setError,
 		clearErrors,
 	} = useForm< CredentialsFormData >( {
 		mode: 'onSubmit',
 		reValidateMode: 'onSubmit',
 		disabled: isPending,
 		defaultValues: {
-			siteAddress: importSiteQueryParam,
```

Comment:
The backend is waiting for a `from_url,` and the front is calling as `siteAddress.` It was causing constantly process of map values to show errors. I just removed it.